schema/pgsql/schema.sql

```diff
@@ -344,89 +362,108 @@ CREATE TABLE rule_escalation (
 
     -- Each position in an escalation can only be used once.
     -- Column position must be NULLed for deletion via "deleted = 'y'"
-    UNIQUE (rule_id, position),
+    CONSTRAINT uk_rule_escalation_rule_id_position UNIQUE (rule_id, position),
 
     CONSTRAINT ck_rule_escalation_not_both_condition_and_fallback_for CHECK (NOT (condition IS NOT NULL AND fallback_for IS NOT NULL)),
-    CONSTRAINT ck_rule_escalation_non_deleted_needs_position CHECK (deleted = 'y' OR position IS NOT NULL)
+    CONSTRAINT ck_rule_escalation_non_deleted_needs_position CHECK (deleted = 'y' OR position IS NOT NULL),
+    CONSTRAINT fk_rule_escalation_rule FOREIGN KEY (rule_id) REFERENCES rule(id),
+    CONSTRAINT fk_rule_escalation_rule_escalation FOREIGN KEY (fallback_for) REFERENCES rule_escalation(id)
 );
 
 CREATE INDEX idx_rule_escalation_changed_at ON rule_escalation(changed_at);
 
 CREATE TABLE rule_escalation_recipient (
     id bigserial,
-    rule_escalation_id bigint NOT NULL REFERENCES rule_escalation(id),
-    contact_id bigint REFERENCES contact(id),
-    contactgroup_id bigint REFERENCES contactgroup(id),
-    schedule_id bigint REFERENCES schedule(id),
-    channel_id bigint REFERENCES channel(id),
+    rule_escalation_id bigint NOT NULL,
+    contact_id bigint,
+    contactgroup_id bigint,
+    schedule_id bigint,
+    channel_id bigint,
 
     changed_at bigint NOT NULL,
     deleted boolenum NOT NULL DEFAULT 'n',
 
     CONSTRAINT pk_rule_escalation_recipient PRIMARY KEY (id),
-
-    CHECK (num_nonnulls(contact_id, contactgroup_id, schedule_id) = 1)
+    CONSTRAINT ck_rule_escalation_recipient_only_one_recipient CHECK (num_nonnulls(contact_id, contactgroup_id, schedule_id) = 1),
```

Comment:
Sorry, but there's still a verb missing here and maybe you want to replace `only` with `exactly` like Alvar suggested. `ck_rule_escalation_recipient_has_exactly_one_recipient`